### PR TITLE
NVPL LAPACK Solver Support on ARM

### PR DESCRIPTION
**Pull request by aayushg55**
* NVPL LAPACK support for all decompositions on ARM

* Simplified SVD interface to hide column-major logic

* More comprehensive type and shape checks for all decompositions

* Batched tests and complex tests for remaining decompositions

* Fixed sign bug in determinant

* Split solver into separate files

Misc:
- Switched all HostExecutor references to const references

Notes:
- The SVD and Eigen methods (gesvd and syevd) are currently not fully optimized in NVPL and give a warning

### CODE REVIEW

File: include/matx/transforms/solver_common.h [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+    default:
+      MATX_ASSERT_STR(false, matxInvalidParameter, "Job for SVD not supported");
+      return '\0';
+  }
+}
+
+/* Solver utility functions */
+
+enum class BatchType {
+  Vector = 1,
```
Comment (cliffburdick) [inline include/matx/transforms/solver_common.h:89]: Should these be VECTOR and MATRIX to be consistent with the others?

File: include/matx/transforms/solver_common.h [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+ * @param tp
+ *   Pointer to pre-allocated memory
+ * @param a
+ *   Tensor to transpose
+ * @param stream
+ *   CUDA stream
+ */
+template <typename TensorType, typename Executor>
+__MATX_INLINE__ auto
+TransposeCopy(typename TensorType::value_type *tp, const TensorType &a, const Executor &exec)
```
Comment (cliffburdick) [inline include/matx/transforms/solver_common.h:154]: This function seems like it should go into tensor_utils.h maybe

File: include/matx/transforms/solver_common.h (hunk 1 above)
Comment (aayushg55) [reply include/matx/transforms/solver_common.h:89]: Yes, will update

File: include/matx/transforms/solver_common.h (hunk 2 above)
Comment (aayushg55) [reply include/matx/transforms/solver_common.h]: Moved it there

**Review by cliffburdick — APPROVED** [review]

**Pull request merged**